playbooks/roles/monitoring-user-defined-projects/README.md

```diff
@@ -0,0 +1,71 @@
+Enable monitoring for user defined projects 
+===========================================
+
+This role enables monitoring services created in the user defined projects. The playbook performs the following steps to enable user defined project monitoring:
+ - Create a new namespace
+ - Deploy a sample prometheus app
+ - Create service to define the monitoring metrics
+ - Deploy a ServiceMonitor resouce object
+ - Update the cluster-monitoring-config object to enable monitoring for user defined projects
+ - Check the service pods running
+ - Grant the permissions to the external user
+ - Extracting the secret token, bearer access token and thanos querier URL
+ - Run querying command to access the monitoring metrics outside the cluster
+
+
+Requirements
+------------
+
+ - Running OCP 4.x cluster.
+
+
+Role Variables
+--------------
+
+| Variable                               | Required | Default                          |Comments                                                                                                                                                             |
+|----------------------------------------|----------|----------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| enable_user_defined_project_monitoring | yes      | false                            | Flag to be set to true to enable monitoring-user-defined-project-monitoring role execution                                                                          |
```

Comment:
Since default is given why is it Required? Variables are Required only when no default is given and is used somewhere when the flag is enabled.

Reply:
The flag has been set with default `false` and mentioned as Required because it requires to be set to `true` and passed as an external argument by user to enable the role execution.

Reply:
> Since default is given why is it Required? Variables are Required only when no default is given and is used somewhere when the flag is enabled.

As the default is given for the flag, the variable has been mentioned as `no` in the Required column. The same has been done for the other variables set with the defaults.